File: javascript_to_golang/experiment10.go
package main

import (
	"fmt"
	"math/rand"
	"time"
)

func findCompany(array []string, input int) {
	for i := 0; i < len(array); i++ {
		if i == input {
			fmt.Printf("Company Found: %s\n", array[input])
			break
		}
		fmt.Printf("Searching Company... %d\n", i+1)
	}
}

func main() {
	rand.Seed(time.Now().UnixNano())
	telco := []string{"Telkom", "Indosat", "XL", "Verizon", "AT&T", "Nippon", "Vodafone", "Orange", "KDDI", "Telefonica", "T-Mobile"}
	company := rand.Intn(11)
	findCompany(telco, company)
}

File: javascript_to_java/experiment-10.java

class experiment10 {
    public static String[] telco = { "Telkom", "Indosat", "XL", "Verizon", "AT&T", "Nippon", "Vodafone", "Orange",
            "KDDI", "Telefonica", "T-Mobile" };

    public static void main(String[] args) {
        int company = (int) Math.round(Math.random() * 10);
        findCompany(telco, company);
    }

    public static void findCompany(String[] array, int input) {
        for (int i = 0; i < array.length; i++) {
            if (array[i] == array[input]) {
                System.out.println("Company Found " + array[input]);
                break;
            }
            System.out.println("Searching Company...");
        }
    }

}
